Lib/test/test_threading.py

```diff
@@ -119,6 +119,13 @@ def func(): pass
             thread = threading.Thread(target=func)
             self.assertEqual(thread.name, "Thread-5 (func)")
 
+    @cpython_only
+    def test_uninitialised_new(self):
+        # Prevent heap types from being created uninitialised (bpo-43916)
+        lock = threading.Lock()
+        tp = type(lock)
+        self.assertRaises(TypeError, tp)
```

Comment:
I'm fine with keeping Python 3.9 behavior, but having to call _thread.allocate_lock() rather than being able to simply call _thread.LockType() is weird. Maybe this API can evolve in Python 3.11.

Reply:
Ok. I'll tear out the threading changes.

Reply:
Please keep this code for now. I'm just thinking aloud.

Reply:
No prob